### PR TITLE
🐞fix for parsing uid spec string

### PR DESCRIPTION
**Pull request by live1206**

### CODE REVIEW

File: src/docfx/build/xref/XrefMap.cs [hunk 1]
```diff
@@ -248,6 +252,13 @@ private static string GetSubstringFromContent(string content, int startLine, int
             var result = new StringBuilder();
             var currentLine = 1;
             var currentColumn = 1;
+
+            // for better performance by accessing index when content is 1 line
+            if (currentLine == startLine && currentLine == endLine)
+            {
+                return content.Substring(startColumn - 1, endColumn - startColumn + 1);
```
Comment (superyyrrzz) [inline src/docfx/build/xref/XrefMap.cs]: Move this to L275? Then this optimization is not limited to currentLine == startLine == endLine **== 1**

Comment (live1206) [reply src/docfx/build/xref/XrefMap.cs]: This only works for **1** line content. If `currentLine == startLine == endLine > 1` we do not know the index directly

**Comment by yufeih** [comment]
@live1206 , what is the performance data for loading `/dotnet` xrefmap before and after JSON reader optimization?

**Comment by live1206** [comment]
> @live1206 , what is the performance data for loading `/dotnet` xrefmap before and after JSON reader optimization?

Original, it was around 48s and now it is around 13s

**Pull request merged**